api/src/test/java/com/epam/pipeline/controller/region/CloudRegionControllerTest.java

```diff
@@ -0,0 +1,460 @@
+
+@WebMvcTest(controllers = CloudRegionController.class)
+public class CloudRegionControllerTest extends AbstractControllerTest {
+
+    private static final long ID = 1L;
+    private static final String TEST_STRING = "TEST";
+    private static final String REGION_URL = SERVLET_PATH + "/cloud/region";
+    private static final String LOAD_PROVIDERS_URL = REGION_URL + "/provider";
+    private static final String LOAD_REGIONS_INFO_URL = REGION_URL + "/info";
+    private static final String LOAD_AVAILABLE_REGIONS_URL = REGION_URL + "/available";
+    private static final String REGION_ID_URL = REGION_URL + "/%d";
+    private AwsRegion awsRegion;
+    private AWSRegionDTO awsRegionDTO;
+    private AzureRegion azureRegion;
+    private AzureRegionDTO azureRegionDTO;
+    private GCPRegion gcpRegion;
+    private GCPRegionDTO gcpRegionDTO;
+    private ResponseResult<AwsRegion> expectedAwsRegionResult;
+    private ResponseResult<AzureRegion> expectedAzureResult;
+    private ResponseResult<GCPRegion> expectedGcpResult;
+
+    @Autowired
+    private CloudRegionApiService mockCloudRegionApiService;
+
+    @Before
+    public void setUp() {
+        awsRegion = RegionCreatorUtils.getDefaultAwsRegion();
+        azureRegion = RegionCreatorUtils.getDefaultAzureRegion();
+        gcpRegion = RegionCreatorUtils.getDefaultGcpRegion();
+
+        awsRegionDTO = RegionCreatorUtils.getDefaultAwsRegionDTO();
+        azureRegionDTO = RegionCreatorUtils.getDefaultAzureRegionDTO();
+        gcpRegionDTO = RegionCreatorUtils.getDefaultGcpRegionDTO();
+
+        expectedAwsRegionResult = ControllerTestUtils.buildExpectedResult(awsRegion);
+        expectedAzureResult = ControllerTestUtils.buildExpectedResult(azureRegion);
+        expectedGcpResult = ControllerTestUtils.buildExpectedResult(gcpRegion);
+    }
+
+    @Test
+    public void shouldFailLoadProvidersForUnauthorizedUser() throws Exception {
+        mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadProviders() throws Exception {
+        final List<CloudProvider> cloudProviders = Collections.singletonList(CloudProvider.AWS);
+
+        Mockito.doReturn(cloudProviders).when(mockCloudRegionApiService).loadProviders();
+
+        final MvcResult mvcResult = mvc().perform(get(LOAD_PROVIDERS_URL)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(EXPECTED_CONTENT_TYPE))
+                .andReturn();
+
+        Mockito.verify(mockCloudRegionApiService).loadProviders();
+
+        final ResponseResult<List<CloudProvider>> expectedResult =
+                ControllerTestUtils.buildExpectedResult(cloudProviders);
+
+        ControllerTestUtils.assertResponse(mvcResult, getObjectMapper(), expectedResult,
```

Comment:
It looks like all the calls to `ControllerTestUtils.assertResponse` in this test class as well as all the others have the same second argument. Moreover in order to use `assertResponse` expected result should be built beforehand using `ControllerTestUtils.buildExpectedResult` method.

I suppose we can introduce overloaded `ControllerTestUtils.assertResponse` method with only three arguments which builds expected result and compares it with an actual result. In order to do that `ControllerTestUtils` can be transformed to a regular bean.

What do you think @ekazachkova? Does it worth trying?

Reply:
@tcibinan  Great idea! Let's implement it.